lib/phoenix/channel.ex

```diff
@@ -338,13 +341,55 @@ defmodule Phoenix.Channel do
     Server.push(transport_pid, topic, event, message, socket.serializer)
   end
 
+  @doc """
+  Replies asynchronously to a socket push.
+
+  Useful when you need to reply to a push that can't otherwise be handled using
```

Comment:
Yes good idea

Sent from my iPhone

> On Oct 31, 2015, at 1:13 PM, José Valim <notifications@github.com> wrote:
> 
> In lib/phoenix/channel.ex:
> 
> > @@ -338,13 +341,55 @@ defmodule Phoenix.Channel do
> >      Server.push(transport_pid, topic, event, message, socket.serializer)
> >    end
> >  
> > +  @doc """
> > +  Replies asynchronously to a socket push.
> > +
> > +  Useful when you need to reply to a push that can't otherwise be handled using
> Should we document that, if you are using tasks, to make sure to pass only the socket_ref? To be more precise, those docs say "how" but not the "why". It would be nice to explain folks why there is a need for socket_ref and why it is a bad idea to pass the socket around.
> 
> —
> Reply to this email directly or view it on GitHub.
>